{
  try {
    taskManager.storeResult(task,e,delegate);
  }
 catch (  Exception inner) {
    inner.addSuppressed(e);
    delegate.onFailure(inner);
  }
}
